app/src/main/java/com/sunasterisk/getitdone/ui/detail/DetailFragment.kt

```diff
@@ -0,0 +1,261 @@
+                )
+            )
+
+            presenter =
+                DetailPresenter(taskListRepository, taskRepository).also { detailPresenter ->
+                    detailPresenter.attach(this)
+                    detailPresenter.getTaskFromId(taskId)
+                    detailPresenter.getAllTaskList()
+                }
+        }
+
+        btnBack.setOnClickListener(this)
+        btnChangeStatus.setOnClickListener(this)
+        btnDelete.setOnClickListener(this)
+
+        imageImportant.setOnClickListener(this)
+
+        myDayLayout.setOnClickListener(this)
+        datePickerLayout.setOnClickListener(this)
+
+        editTextDescription.doOnTextChanged { text, _, _, _ ->
+            text?.let { string ->
+                if (string.isBlank()) imageDescription.setColorFilter(Color.DKGRAY)
+                else task?.let { imageDescription.setColorFilter(Color.parseColor(it.color)) }
+            }
+        }
+    }
+
+    override fun onUpdateTask(isSuccess: Boolean) {
+        var msgId = R.string.msg_update_task_fail
+        if (isSuccess) {
+            setFragmentResult(REQUEST_KEY_UPDATE_TASK, bundleOf(BUNDLE_TASK to task))
+            msgId = R.string.msg_update_task_successfully
+        }
+        displayMessage(msgId)
+        requireActivity().supportFragmentManager.popBackStack()
+    }
+
+    override fun onDeleteTask(isSuccess: Boolean) {
+        var msgId = R.string.msg_delete_task_fail
+        if (isSuccess) {
+            setFragmentResult(REQUEST_KEY_DELETE_TASK, bundleOf(BUNDLE_TASK to task))
+            msgId = R.string.msg_delete_task_successfully
+        }
+        displayMessage(msgId)
+        requireActivity().supportFragmentManager.popBackStack()
+    }
+
+    override fun handleLoadedTaskLists(taskLists: List<TaskList>) {
+        this.taskLists.apply {
+            clear()
+            addAll(taskLists)
+        }
+
+        spinnerTaskLists.adapter = SpinnerArrayAdapter(requireContext(), this.taskLists)
+    }
+
+    override fun handleLoadedTask(task: Task) {
+        this.task = task
+
+        updateUI()
+    }
+
+    override fun handleException(exception: Exception) {
+    }
+
+    override fun onClick(v: View) {
+        when (v.id) {
+            R.id.btnBack -> updateTask()
+            
+            R.id.btnChangeStatus -> {
+                task?.status =
+                    if (task?.status == STATUS_COMPLETED) STATUS_NOT_COMPLETE else STATUS_COMPLETED
+                updateTask()
+            }
+            
+            R.id.btnDelete -> deleteTask()
+
+            R.id.imageImportant -> {
+                task?.let { it.isImportant = !it.isImportant }
+                displayImportantView()
+            }
+
+            R.id.myDayLayout -> {
+                task?.let { it.isInMyDay = !it.isInMyDay }
+                displayInMyDayView()
+            }
+            
+            R.id.datePickerLayout -> {
+                showDateTimePicker()
+            }
```

Comment:
```suggestion
            R.id.datePickerLayout ->   showDateTimePicker()

```
cái trên mình có cmt thì cái dưới cố gắng lm tương tự nhé

Reply:
vâng ạ :facepalm: em có sửa cái khác rồi mà vẫn bị sót cái này